src/test/llmnode_test.cpp

```diff
@@ -0,0 +1,293 @@
+// limitations under the License.
+//*****************************************************************************
+#include <filesystem>
+#include <fstream>
+#include <set>
+#include <sstream>
+#include <string>
+#include <thread>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <openvino/openvino.hpp>
+#include <pybind11/embed.h>
+
+#include "../config.hpp"
+#include "../dags/pipelinedefinition.hpp"
+#include "../grpcservermodule.hpp"
+#include "../http_rest_api_handler.hpp"
+#include "../kfs_frontend/kfs_graph_executor_impl.hpp"
+#include "../kfs_frontend/kfs_grpc_inference_service.hpp"
+#include "../mediapipe_internal/mediapipefactory.hpp"
+#include "../mediapipe_internal/mediapipegraphdefinition.hpp"
+#include "../mediapipe_internal/mediapipegraphexecutor.hpp"
+#include "../metric_config.hpp"
+#include "../metric_module.hpp"
+#include "../model_service.hpp"
+#include "../precision.hpp"
+#include "../python/pythoninterpretermodule.hpp"
+#include "../python/pythonnoderesources.hpp"
+#include "../servablemanagermodule.hpp"
+#include "../server.hpp"
+#include "../shape.hpp"
+#include "../stringutils.hpp"
+#include "../tfs_frontend/tfs_utils.hpp"
+#include "c_api_test_utils.hpp"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#include "mediapipe/framework/calculator_runner.h"
+#pragma GCC diagnostic pop
+
+#include "opencv2/opencv.hpp"
+#include "python/python_backend.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+class LLMFlowTest : public ::testing::TestWithParam<std::string> {
+protected:
+    ovms::Server& server = ovms::Server::instance();
+
+    const Precision precision = Precision::STRING;
+    std::unique_ptr<std::thread> t;
+    std::string port = "9178";
+
+    void SetUpServer(const char* configPath) {
+        ::SetUpServer(this->t, this->server, this->port, configPath);
+    }
+
+    void SetUp() override {
+    }
+    void TearDown() {
+        server.setShutdownRequest(1);
+        t->join();
+        server.setShutdownRequest(0);
+    }
+};
+
+class LLMFlowKfsTest : public LLMFlowTest {
+public:
+    void SetUp() {
+        SetUpServer("/ovms/src/test/llm/config_llm_dummy_kfs.json");
+    }
+};
+
+// --------------------------------------- OVMS LLM nodes tests
+// Test disabled by default - needs LLM models to work in /workspace directory:
+// openvino_detokenizer.bin  openvino_detokenizer.xml  openvino_model.bin  openvino_model.xml  openvino_tokenizer.bin  openvino_tokenizer.xml
+// Model that was used:
+// https://huggingface.co/meta-llama/Llama-2-7b-chat-hf
+// converted with optimum cli
+TEST_F(LLMFlowKfsTest, DISABLED_Infer) {
+    const ovms::Module* grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
+    KFSInferenceServiceImpl& impl = dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
+    ::KFSRequest request;
```

Comment:
The LLM graphs will not be available via KServe API and will **not** accept KServe request. The graphs will be reachable only via OpenAI API. Please check how to set up HTTP OpenAI tests and mock http server streaming request: https://github.com/openvinotoolkit/model_server/pull/2436/files#diff-086971ad0a2af85fdbab17aeff70f49cbeacb9b943246aeb4add720a8312b506

Reply:
I think for now, for the sake of internal testing we could leave it as is. gRPC could be a path for e2e testing until OpenAI is integrated.